spec.md

```diff
@@ -191,12 +191,93 @@ print("Hello, world!")
 # SPDX-SnippetEnd
 ```
 
-### DEP5
+### REUSE.toml
 
-Alternatively, Copyright and Licensing Information MAY be associated with a
-file through a DEP5 file. The intended use case of this method is large
-directories where including a comment header in each file (or in `.license`
-companion files) is impossible or undesirable.
+Copyright and Licensing Information MAY be associated with a file through a
+REUSE.toml file. The intended use case of this method is large directories where
+including a comment header in each file (or in `.license` companion files) is
+impossible or undesirable.
+
+A REUSE.toml file MAY be located in any directory, and can cover files that are
+within its directory or deeper. You MAY have multiple REUSE.toml files in
+different directories.
+
+The `version` key (REQUIRED) MUST have an integer value representing the schema
+version of the file. This specification describes version 1 of REUSE.toml.
+
+Each `[[annotations]]` table represents an association of Copyright and
+Licensing Information to zero or more Covered Files. It has the following keys:
+
+- `path` (REQUIRED), a string or list of strings of paths relative to the
+  REUSE.toml file's directory. Each path is a path to a Covered File with which
+  to associate the table's Copyright and Licensing Information. A path MAY fail
+  to resolve to a file in the Project. A path MAY use globbing. These are the
+  globbing and matching rules:
+
+  - `*` matches everything except slashes, including `.` prefixes.
+
+  - `**` matches everything including slashes.
+
+  - `?` matches any single character.
+
+  - `[seq]` matches any character in the sequence.
+
+  - `[!seq]` matches any character not in the sequence.
```

Comment:
better documentation for this

Reply:
we could rip this functionality out by pre-parsing and escaping square brackets

Reply:
I don't understand your last comment. Better documentation with examples could help.

Reply:
So the problem is that we use the wcglob library in the tool. And the wcglob library has a minimal set of functionality which _includes_ `[seq]` and `[!seq]`. See [wcglob's Syntax](https://facelessuser.github.io/wcmatch/glob/#syntax). For examples of seq, see [here](https://facelessuser.github.io/wcmatch/glob/#posix-character-classes).

What I _don't_ want to do is, in this specification, say 'yeah the globbing rules are just wcglob's behaviour'.

And although I like 99% of the behaviour of wcglob as I have tuned it for the tool, the inclusion of `[seq]` and `[!seq]` is completely unneeded. We _don't need that stuff_, and there's no way to turn it off. But, as you can see from the richness of the examples, documenting that behaviour is hell, and they don't really do that themselves either. They just hold to _longum iter est per praecepta, breve et efficax per exempla_.

There are a few things we can subsequently do:

1. Open an issue against wcglob and plead them to include an option to turn this behaviour off. Or do this ourselves—I can write Python!
2. Do a bad job of documenting this. See: the referenced code.
3. Don't document it at all. This isn't great.
4. Do a good job of documenting this. But: that's a loooot of words spent on a feature _we don't even want_.
5. Make an attempt at turning this behaviour off downstream. Whenever we see `[` or `]`, escape that character internally in the REUSE tool, and the seq behaviour will never be triggered. Subsequently, we needn't document this. But: this may have unintended consequences, maybe. I'm not sure.

Reply:
Created an issue and a PR against wcmatch:

- https://github.com/facelessuser/wcmatch/issues/212
- https://github.com/facelessuser/wcmatch/pull/213

Reply:
Upstream doesn't really want the contribution. I had another look at the code, and I was able do a simple alternative implementation that gets rid of the things we do not want. I will patch this up.

Reply:
The resulting spec is:

```markdown
  - `*` matches everything except forward slashes (i.e. path separators).
  - `**` and `**/` match everything including forward slashes (i.e. path
    separators).
  - In order to escape an asterisk and include it verbatim, prefix it with `\\`.
    You cannot prefix it with `\` because that is invalid TOML. In order to
    include a backslash verbatim, use `\\\\`. `\\` followed by any other
    character is functionally equal to just typing that character.
```

Not _perfect_, especially because TOML means no clean escaping.

This could be simplified if we got rid of escaping. Then asterisks _always_ match.

We could also conditionally insert a backslash on `\\` instead of on `\\\\` if not followed by an asterisk, but that sounds like a bad idea.